core/src/components/input/test/a11y/input.spec.ts

```diff
@@ -0,0 +1,29 @@
+import { newSpecPage } from '@stencil/core/testing';
+import { Input } from '../../input';
+import { Item } from '../../../item/item';
+import { Label } from '../../../label/label';
+
+describe('Input a11y', () => {
+    it('does not set a default aria-labelledby when there is not a neighboring ion-label', async () => {
+        const page = await newSpecPage({
+            components: [Input, Item, Label],
+            html: `<ion-input></ion-input>`
+        })
+
+        const ariaLabelledBy = page.body.querySelector('ion-input > input').getAttribute('aria-labelledby');
+        expect(ariaLabelledBy).toBe(null);
+    });
+
+    it('set a default aria-labelledby when a neighboring ion-label exists', async () => {
+        const page = await newSpecPage({
+            components: [Input, Item, Label],
+            html: `<ion-item>
+                <ion-label>A11y Test</ion-label>
+                <ion-input></ion-input>
+            </ion-item`
+        })
+
+        const ariaLabelledBy = page.body.querySelector('ion-input > input').getAttribute('aria-labelledby');
+        expect(ariaLabelledBy).toBe('ion-input-1-lbl');
```

Comment:
I would prefer if we did not check the exact value here as it is dependent on the number of inputs we use. For example, if we added another test before this that had an input and a label then the attribute value in this test would become `ion-input-2-lbl` which would cause the test to fail. 

You could do something like the following:

```js
const label = page.body.querySelector('ion-label');

...

expect(ariaLabelledBy).toEqual(label.id);
```

I didn't test this, but something along those line should work. We should do the same thing for the other tests as well.